### PR TITLE
feat: add priority queue

### PR DESCRIPTION
**Pull request by kumaryash18**
#### Description of Change
added priority queue in data structure

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Implement priority queue data structure

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/916"></a>

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C/runs/4116112447?check_suite_focus=true) warnings. Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6)! 😃

**Comment by kumaryash18** [comment]
> 
> 
> Hey there! Thank you for submitting a PR! 😄 Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C/runs/4116112447?check_suite_focus=true) warnings. Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6)! 😃

Not sure what to fix.

**Review by Panquesito7 — COMMENTED** [review]
> > Hey there! Thank you for submitting a PR! 😄 Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C/runs/4116112447?check_suite_focus=true) warnings. Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6)! 😃
> 
> Not sure what to fix.

You can see all the warnings in this link https://github.com/TheAlgorithms/C/runs/4116112447?check_suite_focus=true

File: data_structures/queue/priority_queue.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	else {
+		int i;
+		for(i = q.front+1; i <= q.rear; i++)
+			q.queue[i-1] = q.queue[i];
+		q.rear--;
+	}
+	return item;
+}
+
```
Comment (Panquesito7) [inline data_structures/queue/priority_queue.c:81]: Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: data_structures/queue/priority_queue.c (hunk 1 above)
Comment (kumaryash18) [reply data_structures/queue/priority_queue.c:81]: Not Sure how test of this program should look like using `assert`

**Pull request closed without merging**